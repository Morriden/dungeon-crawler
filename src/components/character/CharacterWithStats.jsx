import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import Stats from '../stats/Stats';

const CharacterWithStats = ({ character }) => (
  <section>
    <Character image={character.image}/>
    <Stats 
      health={character.health}
      attack={character.attack}
      energy={character.energy}
      experience={character.energy}
    />
  </section>
);

CharacterWithStats.propTypes = {
  character: PropTypes.shape({
    health: PropTypes.number.isRequired,
    attack: PropTypes.number.isRequired,
    energy: PropTypes.number.isRequired,
    experience: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired
  })
};

export default CharacterWithStats;
