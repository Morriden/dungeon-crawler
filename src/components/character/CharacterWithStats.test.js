import React from 'react';
import { shallow } from 'enzyme';
import CharacterWithStats from './CharacterWithStats';

describe('CharacterWithStats component', () => {
  it('renders CharacterWithStats', () => {
    const character = {
      image: 'im an image',
      health: 100,
      attack: 5,
      energy: 10,
      experience: 0
    };
    const wrapper = shallow(<CharacterWithStats character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
