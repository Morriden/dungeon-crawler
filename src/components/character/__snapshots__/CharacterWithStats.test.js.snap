// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterWithStats component renders CharacterWithStats 1`] = `
<section>
  <Character
    image="im an image"
  />
  <Stats
    attack={5}
    energy={10}
    experience={10}
    health={100}
  />
</section>
`;
